subcommand/connect-init/command.go

```diff
@@ -81,19 +92,84 @@ func (c *Command) Run(args []string) int {
 			return 1
 		}
 	}
+	// First do the ACL Login, if necessary.
+	if c.flagACLAuthMethod != "" {
+		// Validate flags related to ACL login.
+		if c.flagMeta == nil {
+			c.UI.Error("-meta must be set")
+			return 1
+		}
+		if c.flagServiceAccountName == "" {
+			c.UI.Error("-service-account-name must be set")
+			return 1
+		}
+		err = backoff.Retry(func() error {
```

Comment:
This whole section is just scoped inside `if c.flagACLAuthMethod{}` now, functionally it's the same as before but with additional flag validation.